tests/actor/test_actor.py

```diff
@@ -53,6 +53,31 @@ async def _run(self) -> None:
         print(f"{self} done")
 
 
+class RaiseExceptionOnCancelActor(BaseTestActor):
+    """Actor that raises exception during stop."""
+
+    def __init__(
+        self, *, name: str | None = None, recv: Receiver[int], sender: Sender[int]
+    ) -> None:
+        """Create an instance."""
+        super().__init__(name=name)
+        self._recv = recv
+        self._sender = sender
+
+    async def _run(self) -> None:
+        """Start the actor and raise exception after receiving CancelledError."""
+        self.inc_restart_count()
+        if BaseTestActor.restart_count == 1:
+            # This actor should not restart
+            # If it does we just return to avoid infinite await on `stop`
+            return
+        try:
+            async for msg in self._recv:
+                await self._sender.send(msg)
+        except asyncio.CancelledError as exc:
+            raise RuntimeError("Actor should stop.") from exc
```

Comment:
I'm not sure we need to support this, this feels wrong, if a task is cancelled, it should not raise some other exception. Cancellation should always be supported and do nothing if it was already cancelled.